include/dmlc/omp.h

```diff
@@ -8,12 +8,6 @@
 #if defined(_OPENMP)
 #include <omp.h>
 #else
-#ifndef DISABLE_OPENMP
```

Comment:
Why? Is this for non-x86 architectures to not print the message? If so, why not just not print for non-intel architectures?

Reply:
Is so when you compile without openMP you don't get this warning on every compilation unit. Independently of the arch. Personally I always compile without OpenMP. I don't find it ok to have this warning on every file compiled.

Reply:
from ../src/operator/cudnn_algoreg.cc:26:
../dmlc-core/include/dmlc/omp.h:15:81: note: #pragma message: Warning: OpenMP is not available, project will be compiled into single-thread code. Use OpenMP-enabled compiler to get benefit of multi-threading.
                 "Use OpenMP-enabled compiler to get benefit of multi-threading.")
                                                                                 ^
[51/162] Building CXX object CMakeFiles/mxnet_static.dir/src/operator/custom/native_op.cc.o
In file included from ../src/operator/custom/./../../common/utils.h:29:0,
                 from ../src/operator/custom/./../operator_common.h:43,
                 from ../src/operator/custom/./native_op-inl.h:38,
                 from ../src/operator/custom/native_op.cc:26:
../dmlc-core/include/dmlc/omp.h:15:81: note: #pragma message: Warning: OpenMP is not available, project will be compiled into single-thread code. Use OpenMP-enabled compiler to get benefit of multi-threading.
                 "Use OpenMP-enabled compiler to get benefit of multi-threading.")
                                                                                 ^
[52/162] Building CXX object CMakeFiles/mxnet_static.dir/src/operator/softmax_output.cc.o
In file included from ../src/operator/././../common/utils.h:29:0,
                 from ../src/operator/././operator_common.h:43,
                 from ../src/operator/./softmax_output-inl.h:37,
                 from ../src/operator/softmax_output.cc:26:
../dmlc-core/include/dmlc/omp.h:15:81: note: #pragma message: Warning: OpenMP is not available, project will be compiled into single-thread code. Use OpenMP-enabled compiler to get benefit of multi-threading.
                 "Use OpenMP-enabled compiler to get benefit of multi-threading.")
                                                                                 ^
[53/162] Building CXX object CMakeFiles/mxnet_static.dir/src/operator/custom/ndarray_op.cc.o
In file included from ../src/operator/custom/./../../common/utils.h:29:0,
                 from ../src/operator/custom/./../operator_common.h:43,
                 from ../src/operator/custom/./ndarray_op-inl.h:38,
                 from ../src/operator/custom/ndarray_op.cc:26:
../dmlc-core/include/dmlc/omp.h:15:81: note: #pragma message: Warning: OpenMP is not available, project will be compiled into single-thread code. Use OpenMP-enabled compiler to get benefit of multi-threading.
                 "Use OpenMP-enabled compiler to get benefit of multi-threading.")
                                                                                 ^


it's super annoying